tsl/src/compression/compression.c

```diff
@@ -2175,16 +2175,19 @@ create_segment_filter_scankey(RowDecompressor *decompressor, char *segment_filte
 
 /*
  * For insert into compressed chunks with unique index determine the
- * columns which are safe to use for batch filtering.
+ * columns which can be used for INSERT batch filtering.
+ * The passed in relation is the uncompressed chunk.
  *
- * This is based on RelationGetIndexAttrBitmap from postgres with changes
- * to also track unique expression indexes.
+ * In case of multiple unique indexes we have to return the shared columns.
```

Comment:
So this is columns we can safely use for filtering the batches in INSERT to satisfy the UNIQUE checks. The check is based on the unique indexes defined on the uncompressed chunk. These columns might be indexed on the compressed because they are segmentby or have metadata and we can filter on them even if we dont have corresponding compressed index.